process/publisherHandler.go

```diff
@@ -0,0 +1,193 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data"
+	"github.com/multiversx/mx-chain-ws-connector-firehose-go/data/hyperOutportBlocks"
+)
+
+type publisherHandler struct {
+	handler               HyperBlockPublisher
+	outportBlocksPool     BlocksPool
+	dataAggregator        DataAggregator
+	retryDuration         time.Duration
+	firstCommitableBlocks map[uint32]uint64
+
+	blocksChan chan []byte
+	cancelFunc func()
+	closeChan  chan struct{}
+}
+
+// NewPublisherHandler creates a new publisher handler component
+func NewPublisherHandler(
+	handler HyperBlockPublisher,
+	outportBlocksPool BlocksPool,
+	dataAggregator DataAggregator,
+	retryDurationInMiliseconds uint64,
+	firstCommitableBlocks map[uint32]uint64,
```

Comment:
perhaps some checks on the map? not sure what happens if it will be received nil here